File: backend/order-service/src/main/java/com/example/orderservice/orderMenager/api/request/OrderRequest.java
package com.example.orderservice.orderMenager.api.request;

import lombok.*;

import javax.persistence.Column;
import javax.validation.constraints.Future;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.math.BigDecimal;
import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class OrderRequest {
    private Long bikeId;

    private Long accessoryId;

    @Size(max = 5)
    @Column(precision=10, scale=2)
    private BigDecimal price;

    @NotBlank
    private String selectedFrameOption;

    @NotBlank
    @Future
    private Date beginDateOrder;

    @NotBlank
    @Future
    private Date endDateOrder;

    @NotNull
    private Boolean withBikeTrip;
}

File: frontend/src/models/order/request/OrderRequest.ts
export class OrderRequest {
  bikeId: number;
  selectedFrameOption: string;
  beginDateOrder: Date;
  endDateOrder: Date;
  accessoryId: number;
  price: number;
  withBikeTrip: boolean;


  constructor(bikeId: number, selectedFrameOption: string, beginDateOrder: Date, endDateOrder: Date, accessoryId: number, price: number, withBikeTrip: boolean) {
    this.bikeId = bikeId;
    this.selectedFrameOption = selectedFrameOption;
    this.beginDateOrder = beginDateOrder;
    this.endDateOrder = endDateOrder;
    this.accessoryId = accessoryId;
    this.price = price;
    this.withBikeTrip = withBikeTrip;
  }
}
